### DISCUSSION TITLE
Conclusions drawn from Q28/Q29 conflate preference with efficacy/availability

### DISCUSSION THREAD

**Discussion by peter-mlabs** [cardano-foundation/state-of-the-developer-ecosystem#4] [General feedback]
# #28

> The data indicates a preference among developers for real-time, interactive platforms, with Discord servers emerging as the top choice for technical discussions. This preference underscores the value of immediate feedback and a communal problem-solving atmosphere. 

Discord is sometimes the only place to find support. It doesn't imply a preference; I would personally prefer _almost any other platform that allows non-walled-garden searching_, but the other options are anemic and I believe my response was "discord" simply due to this fact.

> Personal networks, including friends, colleagues, and community members, rank significantly, highlighting the importance of interpersonal relationships and peer support in navigating technical challenges.

This may -- and in my experience, does -- indicate that a substantial amount of knowledge in plutus/cardano is "folklore". Documentation is not available, nothing is discoverable. I don't ask colleagues "what hash function is used on chain?" because I _want_ to, I ask them because _the official documentation listed the wrong hash function for over a year".  

# #29

> The insights from this data indicate a community that heavily relies on primary sources, like source code, to understand the technical workings of Cardano. This reflects a demographic that values precise, first-hand information that's free from interpretation or potential inaccuracies. This is a common approach in developer communities, particularly in blockchain and other open-source ecosystems, where the codebase is readily available for scrutiny.

I disagree heavily. A substantial number of my work colleagues, teams I've lead, and members of working groups I am a part of read source _because they have to_, but there is a near-universal consensus that _this should not be the case_. This also holds true for reading the white papers/specs; no one that I've talked to _wants_ to do a spec dive for an answer to a simple question, but we do it often.

> The lower reliance on scientific papers may indicate the dense, often less accessible nature of these materials, especially for developers who are looking for practical, actionable information. However, the fact that they are consulted before social media points to a preference for high-quality information sources over more informal channels, which might be prone to misinformation and speculation.

See above.

> Friends and colleagues ranking above scientific papers, and relatively high overall, suggests the importance of community knowledge sharing and mentorship, reinforcing the idea that the Cardano community is not just technically driven but also socially interconnected.

When I'm leading a team of 10 devs, I don't have time to ask them each to read the spec. People dive deep where they need to and then communicate it as folklore. This conclusion is, again, not supported by the the data.

**Comment by KtorZ** [comment]
I am happy to take any concrete suggestions to improve the current formulation. Though, you also seem to also conflate interpretations of the current results with recommendations that this is the way to go. The report is in no-way a manual/recipe on how things _should be_ but rather just an observation at a point in time. 

I agree with you when you say that reading source code ranks that high because people have to. Yet, Cardano (Shelley) has been around for 3.5 years and no one really managed to successfully put this knowledge in a different form. So perhaps, people do still prefer to read code after all because any attempt to provide that knowledge otherwise as failed?

Same for Discord, it becomes the de-facto place to ask questions despite being the preferred approach. Yet we have a forum, a stack exchange, github discussions, CIPs, working groups, etc... Why then everyone keep asking questions on Discord?


Interpretation is hard and subjective. We can reformulate / put more nuance in few of them based on your feedback. I'd appreciate however _concrete reformulations_ or additions. Keep in mind though that we won't likely ever get to an interpretation that satisfy everyone. Which is why we also provide the raw data. I encourage anyone who has time to do a better / deeper analysis to do so and to contribute back. I had little time.

**Reply by peter-mlabs** [reply]
Thanks for the response, @KtorZ! 

For context, I am part of a working group that is trying to determine where to allocate resources to improve developer experience. This survey was pointed to as something that both ourselves and funding entities can (and will) look at in order to guide that allocation. This is where I think the danger comes in: taking observational data and drawing preferential conclusions from it leads to entrenching the status quo. 

For something like moving away from spec-and-source-code as primary sources of information, this is especially important. _"Do perhaps, people do still prefer to read code after all because any attempt to provide that knowledge otherwise as failed?"_ is exactly right -- it has failed, because the system is complex and disorganized. We need funding for technical writers and editors; we need buy-in from the community to censor or deprecate out-of-date, unofficial resources; we need official specs and source code to provide stable links to the tutorials; I need clients to understand that the documentation situation is a slow down and be more eager to contribute joint funding to improvement efforts. That sort of buy-in becomes harder if the official line from the CF is "devs prefer reading source code".

WRT analysis, it's great that the full results are correlated per user! This will be a big help. We still can't treat it as a randomly sampled survey, but the data collect here is still better than anything else we've got. To that end, I've written a JQ snippet that will transform the results into a tabular, CSV format per user and separate multi-choice answers per string. I'm also working on some python/pandas to split the multi-choice responses into individual columns for further analysis. 

```jq
keys_unsorted as $k | map(to_entries) | map(map({respondent: .key, answer: .value})) | flatten | group_by(.respondent) | map(map(.answer)) as $answers | [$k] + $answers | .[] | map((if type=="array" then join("|") else . end)) | @csv
```

----

In terms of reformulations of the interpretations of these specific questions, I think I would prefer to see the information from Q30 reflected more in the interpretation of Q28/Q29. 

Let me do some additional analysis and see if there's a meaningful correlation between satisfaction and any of the documentation categories -- a strong correlation between a low satisfaction and reliance on source code, for example, would support my hypothesis; high satisfaction <> discord would support yours, and these two are not necessarily mutually exclusive. We can't completely quantify it with it being multiple choice (high satisfaction with discord, low satisfaction with source cord might result in a lukewarm "6" overall), but there might be strong trends hiding.

Does this sound acceptable to you? Happy to contribute back any code we use to do the analysis for both inspection and future adaptation
